src/infill/LightningTreeNode.cpp

```diff
@@ -406,7 +406,7 @@ void LightningTreeNode::removeJunctionOverlap(Polygons& result_lines, const coor
         PolygonRef polyline = *poly_it;
         if (polyline.size() <= 1)
         {
-            polyline = std::move(result_lines.back());
+            polyline = static_cast<PolygonRef&&>(result_lines.back());
```

Comment:
I have to give this line a little extra thought tbh. But that is mostly because our `PolygonRef` class makes so little sense to me.  As far as I remember it is a sort of wrapper with references to the polygon, casting this to a `refref`  seems strange to me. But that is also partly because I watched this YouTube video last week https://www.youtube.com/watch?v=2Y9XbltAfXs and around 38minutes  he talks about Template argument deduction, he shows a table what the deduced argument would be and passing a `T&` to a `T&&` will still result in `T&`.

Reply:
If I recall correctly, both the above line are equivalent. I don't remember exactly why I did that, maybe the sole purpose was to silence a warning.
`PolygonRef` is a non owning pointer wrapper, so copying it ([like here](https://github.com/Ultimaker/CuraEngine/pull/1582/files#diff-0b8a0d9d420d4449bc50f348e965e7b418df3b843bc48572a58ceb2aee0eacedL1254-R1254)) shouldn't be an issue.

Reply:
In the end I remove the moves.